### ISSUE TITLE
[match] Unexpected errored build when using s3 storage option within fastlane match

### ISSUE THREAD

**Issue by AustinFuller** [fastlane/fastlane#16322]
# New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
While using `s3` as the `storage_mode`, an unexpected warning from the `aws-sdk-java` dependency causes a blank error to halt the lane execution. 

##### Command executed
Before running any context of pilot or build_ios_app, I set up my provisioning profiles using match. I run: 
```
lane :setup do
	if is_ci?
		setup_ci
		sh('bundle exec fastlane fastlane-credentials add --username $DEPLOYMENT_USER --password $DEPLOYMENT_PASSWORD')
	end

	match(
            storage_mode: 's3',
            s3_bucket: ...,
            s3_region: 'us-east-1',
            s3_access_key: $AWS_ACCESS_KEY_ID,
            s3_secret_access_key: $AWS_SECRET_ACCESS_KEY,
            verbose: true
         )
end
```

The output of interest is: 
```
+----------------------------+-----------------------------------------------------------------------+
|                                     Summary for match 2.145.0                                      |
+----------------------------+-----------------------------------------------------------------------+
| storage_mode               | s3                                                                    |
| s3_bucket                  | ...                                           |
| s3_region                  | us-east-1                                                             |
| verbose                    | true                                                                  |
| type                       | appstore                                                              |
| readonly                   | false                                                                 |
| generate_apple_certs       | true                                                                  |
| skip_provisioning_profiles | false                                                                 |
| app_identifier             | ["...", "...",      |
|                            | "...", "...",  |
|                            | "..."]                                       |
| username                   | ...                                               |
| team_name                  | ...                                                          |
| git_branch                 | master                                                                |
| shallow_clone              | false                                                                 |
| clone_branch_directly      | false                                                                 |
| keychain_name              | login.keychain                                                        |
| force                      | false                                                                 |
| force_for_new_devices      | false                                                                 |
| skip_confirmation          | false                                                                 |
| skip_docs                  | false                                                                 |
| platform                   | ios                                                                   |
+----------------------------+-----------------------------------------------------------------------+

S3 client configured for "us-east-1" but the bucket "..." is in "us-east-1"; Please configure the proper region to avoid multiple unnecessary redirects and signing attempts
WARN [2020-04-16 12:10:53.88]: Lane Context:
INFO [2020-04-16 12:10:53.88]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>nil, :LANE_NAME=>"deployProduction"}
ERROR [2020-04-16 12:10:53.88]: 
INFO [2020-04-16 12:10:53.88]: Successfully generated documentation at path '/Users/austinfuller/Development/hitchpin-ios/fastlane/README.md'

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 3    | Switch to setup lane       | 0           |
| 4    | is_ci                      | 0           |
| 💥   | match                      | 0           |
+------+----------------------------+-------------+

DEBUG [2020-04-16 12:10:53.89]: All plugins are up to date
ERROR [2020-04-16 12:10:53.89]: fastlane finished with errors
```

Most notably: `S3 client configured for "us-east-1" but the bucket "..." is in "us-east-1"; Please configure the proper region to avoid multiple unnecessary redirects and signing attempts` and `ERROR [2020-04-16 12:10:53.88]: `

From what I've gathered is that the S3 client configuration is just a warning from the aws sdk rather than a hard error. The best I've been able to run up after reading around is [here](https://github.com/aws/aws-sdk-java/issues/1338).

### In short
I think that the aws-sdk-java is throwing a warning that is being handled like an error stopping execution of the lane and forwarding a blank error.  The expected behavior is the S3 client configuration warning to not occur (issue lives within the dependency so outside the scope of fastlane) or that a blank error doesn't stop lane execution. 

### Environment
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.15.4                                                 |
| Ruby                        | 2.5.8                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.21.1 (Apple Git-122.3)                    |
| Installation Source         | /usr/local/Cellar/fastlane/2.145.0/libexec/bin/fastlane |
| Host                        | Mac OS X 10.15.4 (19E266)                               |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.5/2.5.8/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1f  31 Mar 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 11.3.1                                                  |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile (with some redactions) `</summary>

```ruby
# frozen_string_literal: true

fastlane_version '2.145.0'

default_platform :ios

slack_url = ...

lane :deployProduction do |_options|
	setup

	build_ios_app(
		workspace: '....xcworkspace',
		configuration: 'Release',
		scheme: '...',
		clean: false,
		include_symbols: true,
		include_bitcode: true,
		silent: true,
		export_xcargs: '-allowProvisioningUpdates'
	)

	slack(
		slack_url: slack_url,
		message: 'Successfully uploaded production release to Appstore Connect!'
	)

	deliver(
		username: "...",
		submit_for_review: true,
		force: true,
		app_identifier: '...',
		metadata_path: './metadata'
	)

    instabug_official(api_token: '***', dsym_array_paths:[ ])
end

lane :deployStaging do |_options|
	setup

	deploy(
		config: 'Staging',
		app_identifier: '....staging',
		dsym_path: './....app.dSYM.zip'
	)
    instabug_official(api_token: '***', dsym_array_paths:[ ])
end

lane :deployTesting do |_options|
	setup

	deploy(
		config: 'Testing',
		app_identifier: '....testing',
		dsym_path: './....app.dSYM.zip'
	)
end


lane :deployVIP do |_options|
	setup

	deploy(
		config: 'VIP',
		app_identifier: '....VIP',
		dsym_path: './....app.dSYM.zip'
	)
end

lane :deployDemo do |_options|
	setup

	deploy(
		config: 'Demo',
		app_identifier: '....demo',
		dsym_path: './....app.dSYM.zip'
	)
end

lane :setup do
	if is_ci?
		setup_ci
		sh('bundle exec fastlane fastlane-credentials add --username $DEPLOYMENT_USER --password $DEPLOYMENT_PASSWORD')
	end

	match(
        storage_mode: 's3',
        s3_bucket: '...',
        s3_region: 'us-east-1',
        s3_access_key: $AWS_ACCESS_KEY_ID,
        s3_secret_access_key: $AWS_SECRET_ACCESS_KEY,
        verbose: true
    )

end

lane :deploy do |options|
	build_ios_app(
		workspace: '....xcworkspace',
		configuration: options[:config],
		scheme: '...',
		clean: false,
		include_symbols: true,
		include_bitcode: true,
		silent: true,
		export_xcargs: '-allowProvisioningUpdates'
	)

	pilot(
		username: '...',
		app_identifier: options[:app_identifier],
		beta_app_feedback_email: '...',
		changelog: 'Incremental Build of HitchPin iOS!',
		notify_external_testers: true,
		team_name: '...'
	)

	slack(
		slack_url: slack_url,
		message: "Successfully uploaded to TestFlight: #{options[:app_identifier]}"
	)
end

lane :runTests do
	scan(
		scheme: '...',
		workspace: '....xcworkspace',
		code_coverage: true,
        slack_url: slack_url,
		sdk: 'iphonesimulator',
        derived_data_path: "DerivedData"
	)
end

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.145.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                            | Version | Update-Status |
| --------------------------------- | ------- | ------------- |
| fastlane-plugin-instabug_official | 0.2.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                               | Version      |
| --------------------------------- | ------------ |
| slack-notifier                    | 2.3.2        |
| atomos                            | 0.1.3        |
| CFPropertyList                    | 3.0.2        |
| claide                            | 1.0.3        |
| colored2                          | 3.1.2        |
| nanaimo                           | 0.2.6        |
| xcodeproj                         | 1.15.0       |
| rouge                             | 2.0.7        |
| xcpretty                          | 0.3.0        |
| terminal-notifier                 | 2.0.0        |
| unicode-display_width             | 1.7.0        |
| terminal-table                    | 1.8.0        |
| plist                             | 3.5.0        |
| public_suffix                     | 2.0.5        |
| addressable                       | 2.7.0        |
| multipart-post                    | 2.0.0        |
| word_wrap                         | 1.0.0        |
| tty-screen                        | 0.7.1        |
| tty-cursor                        | 0.7.1        |
| tty-spinner                       | 0.9.3        |
| babosa                            | 1.0.3        |
| colored                           | 1.2          |
| highline                          | 1.7.10       |
| commander-fastlane                | 4.4.6        |
| excon                             | 0.73.0       |
| faraday                           | 0.17.3       |
| unf_ext                           | 0.0.7.7      |
| unf                               | 0.1.4        |
| domain_name                       | 0.5.20190701 |
| http-cookie                       | 1.0.3        |
| faraday-cookie_jar                | 0.0.6        |
| faraday_middleware                | 0.13.1       |
| fastimage                         | 2.1.7        |
| gh_inspector                      | 1.1.3        |
| json                              | 2.3.0        |
| mini_magick                       | 4.10.1       |
| multi_xml                         | 0.6.0        |
| rubyzip                           | 1.3.0        |
| security                          | 0.1.3        |
| xcpretty-travis-formatter         | 1.0.0        |
| dotenv                            | 2.7.5        |
| bundler                           | 2.1.4        |
| naturally                         | 2.2.0        |
| simctl                            | 1.6.8        |
| jwt                               | 2.1.0        |
| uber                              | 0.1.0        |
| declarative                       | 0.0.10       |
| declarative-option                | 0.1.0        |
| representable                     | 3.0.4        |
| retriable                         | 3.1.2        |
| mini_mime                         | 1.0.2        |
| multi_json                        | 1.14.1       |
| signet                            | 0.14.0       |
| memoist                           | 0.16.2       |
| os                                | 1.1.0        |
| googleauth                        | 0.11.0       |
| httpclient                        | 2.8.3        |
| google-api-client                 | 0.36.4       |
| google-cloud-env                  | 1.3.1        |
| google-cloud-errors               | 1.0.0        |
| google-cloud-core                 | 1.5.0        |
| digest-crc                        | 0.5.1        |
| google-cloud-storage              | 1.25.1       |
| emoji_regex                       | 1.0.1        |
| jmespath                          | 1.4.0        |
| aws-partitions                    | 1.294.0      |
| aws-eventstream                   | 1.0.3        |
| aws-sigv4                         | 1.1.1        |
| aws-sdk-core                      | 3.92.0       |
| aws-sdk-kms                       | 1.30.0       |
| aws-sdk-s3                        | 1.61.2       |
| fastlane-plugin-instabug_official | 0.2.1        |
</details>

*generated on:* **2020-04-16**
</details>

**Comment by AustinFuller** [comment]
Relevant PRs include: 
https://github.com/fastlane/fastlane/pull/15200
https://github.com/fastlane/fastlane/pull/16143

**Comment by AustinFuller** [comment]
Not sure if related, but I make no use of `fastlane-plugin-s3` for distribution. I run everything through TestFlight.

**Comment by AustinFuller** [comment]
I've read into the stack trace more and I think that this is outside the scope of fastlane actually. Trying to provide a stack trace.

**Comment by AustinFuller** [comment]
Generating the repo on 2.143.0 (which dependent on v2 of aws-sdk) makes this error go away even when using 2.145.0 later. 

Honestly, I'm quite lost on getting this to work and making this issues concise and actionable, but leaving open in case someone like @joshdholtz happens to know something specific and not documented. 

Josh, if you were available for a quick chat, I'm experiencing a lot of pain using match with s3 and have some questions that could make it easier. 

Thanks.

**Comment by joshdholtz** [comment]
@AustinFuller I’d love to help!  I’ll try to take a look through this history first but want to DM me on Twitter at `@joshdholtz`?

**Comment by AustinFuller** [comment]
Josh is awesome, ping him on twitter and we nailed down some questions I had. 

While I'm not certain in this, I think the s3 bucket issue above is more an error with the access key that I'm failing to pass in from Github Action's secrets. I think that the environment variable setup isn't the same / is a little different so its throwing everything off there. 

Closing as I can't responsible reproduce the error now.

If you reach this issue after excessive Googling, feel free to reach out to me on Twitter `@__austinfuller` and I'll explain the secrets / environment variable issue I was having and how I could it.
